### PR TITLE
KT-4748

### PR DESCRIPTION
**Pull request by Dimach**
Inspection and tests.

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/DoubleNegationInspection.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                        return
+                    }
+                    var parent = expression.parent
+                    while (parent is KtParenthesizedExpression) {
+                        parent = parent.parent
+                    }
+                    if (parent is KtPrefixExpression && parent.operationToken == KtTokens.EXCL) {
+                        holder.registerProblem(expression,
+                                               "Redundant double negation",
+                                               ProblemHighlightType.WEAK_WARNING,
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/DoubleNegationInspection.kt]: For future: use here `GENERIC_ERROR_OR_WARNING` if you want to get highlight type depending on inspection severity level (by default, from plugin.xml, or configured by user). Explicitly given `WEAK_WARNING` enforces this severiry level making it unconfigurable by user (will fix myself).

**Comment by mglukhikh** [comment]
Looks good, I will test this inspection.

File: idea/src/org/jetbrains/kotlin/idea/inspections/DoubleNegationInspection.kt (hunk 1 above)
Comment (Dimach) [inline idea/src/org/jetbrains/kotlin/idea/inspections/DoubleNegationInspection.kt]: Ok, thanks for tip.

**Comment by mglukhikh** [comment]
Also, inspection description was forgotten, added (hint: run InspectionDescriptionsTest). Merged manually into master, thank you.

**Pull request closed without merging**